src/databricks/labs/ucx/azure/resources.py

```diff
@@ -374,3 +410,86 @@ def managed_identity_client_id(
                 return None
             return principal.client_id
         return None
+
+    def get_access_connector(self, subscription_id: str, resource_group_name: str, name: str) -> AccessConnector:
```

Comment:
move these new methods to `AzureResources` class (higher level), which calls `AzureApi` class (lower level) and reduce copy-paste:
https://github.com/databrickslabs/ucx/blob/5563b661eafaab71fd6233153094c513c6ee76de/src/databricks/labs/ucx/azure/resources.py#L481-L512

Reply:
These methods are already in AzureResources added under `AzureResources.managed_identity_client_id`